### PR TITLE
[New Check]: TimeSeries with empty dataset

### LINKED ISSUE #324: [Add Check]: Datasets are not empty
### What would you like to see added to the NWBInspector?

From https://github.com/NeurodataWithoutBorders/pynwb/issues/1002, check that datasets are not empty.

Maybe need to consider edge case of ImageSeries with external file format, memory doesn't reclal if the `Dataset` there is `None`, empty, or if the `data` attribute is not set at all in that case.

### Pseudocode or other logic of the check

_No response_

### Do you have any interest in helping implement the check function?

Yes.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/NeurodataWithoutBorders/nwbinspector/blob/dev/.github/CODE_OF_CONDUCT.rst)
- [X] Have you ensured this check was not already [requested](https://github.com/neurodatawithoutborders/nwbinspector/issues)?

### PR DESCRIPTION
**Pull request by alessandratrapani**
Related to Issue #324 
add check function for empty dataset

- [x] add check function
- [ ] add unit tests
- [ ] add check function documentation 
- [ ] update CHANGELOG.md

### CODE REVIEW

**Comment by CodyCBakerPhD** [comment]
@alessandratrapani Very minor thing, but it seems as if GitHub is not smart enough to recognize the keyphrase 'Related to Issue #...' when automatically adding the development issue link

I always do 'fix #...', I think I've seen 'resolve #...' work for others

No worries, though - it's easy to add it afterward through the web interface

File: src/nwbinspector/checks/nwb_containers.py [hunk 1]
```diff
@@ -28,6 +28,25 @@ def check_large_dataset_compression(nwb_container: NWBContainer, gb_lower_bound:
...
+@register_check(importance=Importance.BEST_PRACTICE_VIOLATION, neurodata_type=NWBContainer)
+def check_dataset_is_empty(nwb_container: NWBContainer):
+
+    def is_empty_data(data):
+        if isinstance(data, ndarray):
+            return data.size == 0
+        elif isinstance(data, (list, tuple)):
+            return all(is_empty_data(item) for item in data)
+        else:
+            return not bool(data)
```
Comment (CodyCBakerPhD) [inline src/nwbinspector/checks/nwb_containers.py:40]: I wouldn't mind if this were moved to the common `utils` to be used by any other check (or anyone else) going forward 😃

**Comment by CodyCBakerPhD** [comment]
I want to say the basic structure looks good for checking the `data` field; there are perhaps some other cases we must consider

The best way to tell will be to add several tests (cases where it should pass without raising a message, cases where it should catch 'emptiness', skips on certain edge cases such as `ImageSeries`)

**Comment by h-mayorquin** [comment]
Closing now that https://github.com/NeurodataWithoutBorders/nwbinspector/pulls?q=is%3Apr+is%3Aclosed is merged
